Switch from js-yaml to yamljs

mbland opened the pull request:
Turns out, per #3, that js-yaml isn't browserify-friendly. It appears that yamljs is. However, this commit currently causes a test failure:

```
  6 passing (33ms)
  1 failing

  1) validate should return an array of errors if the YAML fails to parse:
     AssertionError: expected undefined not to be empty
      at Context.<anonymous> (test/validator-test.js:48:32)
```

The following payload should fail because strings that contain `:` should be wrapped in quotes:

``` js
      var yamlContents = 'lyric: watch out boy: she\'ll chew you up!',
```

But validation passes because yamljs is a little too smart, and will figure out how to wrap such malformed strings:

``` js
{ lyric: 'watch out boy: she\'ll chew you up!' }
```

**Question: Accept this robust behavior, and find another parse error to use in the test, or try finding another library that is as strict as js-yaml?**

**Question the second:** @jeremiak can you post, either here or in #3, the commands/snippets you used to check the browserification?

cc: @ccostino

Commit:
Switch from js-yaml to yamljs

Turns out, per #3, that js-yaml isn't browserify-friendly. It appears that
yamljs is. However, this commit currently causes a test failure:

```
  6 passing (33ms)
  1 failing

  1) validate should return an array of errors if the YAML fails to parse:
     AssertionError: expected undefined not to be empty
      at Context.<anonymous> (test/validator-test.js:48:32)
```

The following payload should fail because strings that contain `:` should be
wrapped in quotes:

```js
      var yamlContents = 'lyric: watch out boy: she\'ll chew you up!',
```

But validation passes because yamljs is a little too smart, and will figure
out how to wrap such malformed strings:

```js
{ lyric: 'watch out boy: she\'ll chew you up!' }
```

Question: Accept this robust behavior, and find another parse error to use in
the test, or try finding another library that is as strict as js-yaml?

ccostino commented:
Counter-question to your initial question @mbland:  Is this robust behavior either configurable or something that we want?  If the answer is yes (and in doing a cursory glance, it's not configurable without modifying the library, so bear that in mind), then I say we go with this change if it buys us this nice added bonus and update the test accordingly.  If the answer is no, then it's back to the drawing board for us.

My vote is yes, I'm fine with the added robustness and I like that this will let us browserify the module.  This will enable us to use it in the user-facing tools, which I think will make life a lot easier for folks and lower another barrier for wider adoption.

jeremiak commented:
@mbland - as it is currently the only thing you need to do is require the parser and instantiate the class:

```js
var Validator = require('about-npm-yml');
var validator = new Validator();
```

and when you try and run `browserify` against that you'll get an error in the resulting bundle.

I personally would be ok with accepting the more robust behavior, especially for now since we haven't seen too many cases of needed really strict parsing.

Commit:
Fix malformed YAML test by introducing new error

Per the consensus from #4, we're keeping the more robust YAML parsing behavior
from yamljs.

Commit:
Fix `npm run-script lint` error

mbland commented:
OK gang, pushed a new commit to fix the test (and another to fix a lint error).

ccostino commented:
Looks great to me!  Will wait for @jeremiak to take a quick look.

jeremiak commented:
Looks good to me, sorry for taking a while to get to this. I'll merge it now
